crates/scheduler/src/scheduling/batch_scheduler.rs

```diff
@@ -221,267 +396,154 @@ impl BatchScheduler {
+                },
+                price: PriceRange::default(),
+                min: 0,
+                target: 0,
+                grace: Duration::from_secs(1),
+            },
+        );
+        let worker_pool = PoolWithStatistics::<RunningMean>::new(pool);
+        let worker_handle = worker_pool.handle();
+        let ps_pool = Pool::new(
+            NoopAllocator,
+            PoolConfig {
+                name: "ps".into(),
+                spec: hypha_messages::WorkerSpec {
+                    resources: Resources::default(),
+                    executor: vec![],
+                },
+                price: PriceRange::default(),
+                min: 0,
+                target: 0,
+                grace: Duration::from_secs(1),
+            },
+        );
+        let parameter_pool = ps_pool.handle();
 
-        let steps = vec![
-            Step::new(
-                w3,
-                Status(progress::Status { batch_size: 50 }),
-                Response::Continue,
-                500,
-            ),
-            Step::new(
-                w2,
-                Status(progress::Status { batch_size: 100 }),
-                Response::Continue,
-                300,
-            ),
-            Step::new(
-                w1,
-                Status(progress::Status { batch_size: 150 }),
-                Response::Continue,
-                150,
-            ),
-            Step::new(
-                w3,
-                Status(progress::Status { batch_size: 50 }),
-                Response::ScheduleUpdate { counter: 2 },
-                50,
-            ),
-            Step::new(
-                w3,
-                Status(progress::Status { batch_size: 50 }),
-                Response::Continue,
-                500,
-            ),
-            Step::new(
-                w2,
-                Status(progress::Status { batch_size: 100 }),
-                Response::ScheduleUpdate { counter: 1 },
-                100,
-            ),
-            Step::new(
-                w1,
-                Status(progress::Status { batch_size: 150 }),
-                Response::ScheduleUpdate { counter: 0 },
-                300,
-            ),
-            Step::new(
-                w3,
-                Status(progress::Status { batch_size: 50 }),
-                Response::Continue,
-                100,
-            ),
-            Step::new(
-                w2,
-                Status(progress::Status { batch_size: 100 }),
-                Response::Continue,
-                100,
+        let (tx, _rx) = tokio::sync::mpsc::channel::<(PeerId, Metrics)>(1);
+        let round = std::sync::Arc::new(tokio::sync::Mutex::new(RoundState::default()));
+        let resp = schedule::<RunningMean, BasicSimulation>(
+            tx,
+            worker_handle,
+            parameter_pool,
+            round,
+            std::time::Instant::now(),
+            (
+                PeerId::random(),
```

Comment:
I understand why it is this way, but we probably don't want to accept messages from a worker that is not in the pool.